pandas/tests/io/parser/compression.py

```diff
@@ -166,6 +166,21 @@ def test_read_csv_infer_compression(self):
 
         inputs[3].close()
 
+    def test_read_csv_infer_compression_pathlib(self):
+        """
+        Test that compression is inferred from pathlib.Path paths.
+        """
+        try:
+            import pathlib
+        except ImportError:
+            pytest.skip('need pathlib to run')
+        expected = self.read_csv(self.csv1, index_col=0, parse_dates=True)
```

Comment:
I'm not sure how to do that... I followed the method used by other tests in this module, e.g. `test_read_csv_infer_compression`. Is `csv1` available in memory somewhere? I agree that it would be nice to reduce the runtime burden of this test / dependencies on a file... But, it's not something I should manually create, I don't think.